hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -452,10 +453,10 @@ && noSubsequentReplaceCommit(earliestInstant.requestedTime(), partitionPath)) {
    */
   private boolean hasPendingFiles(String partitionPath) {
     try {
-      HoodieTableFileSystemView fsView = HoodieTableFileSystemView.fileListingBasedFileSystemView(context, hoodieTable.getMetaClient(), hoodieTable.getActiveTimeline());
       StoragePath fullPartitionPath = new StoragePath(hoodieTable.getMetaClient().getBasePath(), partitionPath);
-      fsView.addFilesToView(partitionPath, FSUtils.getAllDataFilesInPartition(
-          hoodieTable.getStorage(), fullPartitionPath));
+      List<StoragePathInfo> allDataFilesInPartition = FSUtils.getAllDataFilesInPartition(
```

Comment:
Did you see the comment above, we can not use the MDT files fetching here.

Reply:
This is the same code that exists today. `FSUtils.getAllDataFilesInPartition` uses file listing and not MDT.

I am just using the proper constructor here for when you want to construct a view from a set of files. This allows us to keep the `addFilesToView` private.

Reply:
okay...